### PR TITLE
PR: Add an option to sort editor tabs alphabetically

### PR DESCRIPTION
**Pull request by AlexDeweert**
## Description of Changes

* [X] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [X] Included a screenshot (if affecting the UI)

Added a small feature that sorts the editor tabs alphabetically. Sometimes having multiple files open at the same time can become overwhelming to sort through to find a file of interest to continue editing or to reference it contents, and although the file explorer can simply be used to re-select the file, sometimes the file explorer is navigated away from the location of the other files.

A menu item has been added to the editor tabs "option" menu at the top right corner of the editor tab widget stack.

sort_tabs_1

sort_tabs_2

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Alexander L. Deweert

### CODE REVIEW

File: spyder/plugins/editor/widgets/editor.py [hunk 1]
```diff
@@ -1568,6 +1571,20 @@ def close_all_but_this(self):
         for i in range(0, self.get_stack_count()-1  ):
             self.close_file(0)
 
+    """ Sort open tabs alphabetically """
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py]: Please move this docstring to be inside the function definition (just below `def sort_file_tabs(self)`). Also remove the spaces around the text and add a dot at the end.

File: spyder/plugins/editor/widgets/editor.py [hunk 2]
```diff
@@ -1568,6 +1571,20 @@ def close_all_but_this(self):
...
             self.close_file(0)
 
+    """ Sort open tabs alphabetically """
+    def sort_file_tabs(self):
+        while self.sorted() == False:
+            for i in range(0,self.tabs.tabBar().count()):
+                if( self.tabs.tabBar().tabText(i) > self.tabs.tabBar().tabText(i+1) ):
+                    self.tabs.tabBar().moveTab(i, i+1)
+
+    """ Utility function for sort_file_tabs """
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py]: Please also do the same for this docstring.

**Comment by ccordoba12** [comment]
@CAM-Gerlach, could you review the functionality of this PR? Thanks!

File: spyder/plugins/editor/widgets/editor.py (hunk 1 above)
Comment (AlexDeweert) [reply spyder/plugins/editor/widgets/editor.py]: Made change

File: spyder/plugins/editor/widgets/editor.py (hunk 2 above)
Comment (AlexDeweert) [reply spyder/plugins/editor/widgets/editor.py]: Made change

**Comment by AlexDeweert** [comment]
@CAM-Gerlach I'm seeing two failures but I'm not sure why exactly. Any help would be appreciated.

Capture

File: spyder/plugins/editor/widgets/editor.py [hunk 3]
```diff
@@ -1568,6 +1571,22 @@ def close_all_but_this(self):
         for i in range(0, self.get_stack_count()-1  ):
             self.close_file(0)
 
+    def sort_file_tabs_alphabetically(self):
+        """Sort open tabs alphabetically."""
+        while self.sorted() is False:
+            for i in range(0, self.tabs.tabBar().count()):
+                if(self.tabs.tabBar().tabText(i) >
+                   self.tabs.tabBar().tabText(i+1)):
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py]: This needs to be

```python
if (self.tabs.tabBar().tabText(i) >
        self.tabs.tabBar().tabText(i + 1)):
```

The second line is indented 8 spaces with respect to the `if`.

File: spyder/plugins/editor/widgets/editor.py [hunk 4]
```diff
@@ -1568,6 +1571,22 @@ def close_all_but_this(self):
...
+            for i in range(0, self.tabs.tabBar().count()):
+                if(self.tabs.tabBar().tabText(i) >
+                   self.tabs.tabBar().tabText(i+1)):
+                    self.tabs.tabBar().moveTab(i, i+1)
+
+    def sorted(self):
+        """Utility function for sort_file_tabs_alphabetically()."""
+        for i in range(0, self.tabs.tabBar().count()-1):
+            if (self.tabs.tabBar().tabText(i) >
+                self.tabs.tabBar().tabText(i + 1)):
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py]: This needs to be

```python
if (self.tabs.tabBar().tabText(i) >
        self.tabs.tabBar().tabText(i + 1)):
```

File: spyder/plugins/editor/widgets/editor.py [hunk 5]
```diff
@@ -1568,6 +1571,22 @@ def close_all_but_this(self):
...
+        """Sort open tabs alphabetically."""
+        while self.sorted() is False:
+            for i in range(0, self.tabs.tabBar().count()):
+                if(self.tabs.tabBar().tabText(i) >
+                   self.tabs.tabBar().tabText(i+1)):
+                    self.tabs.tabBar().moveTab(i, i+1)
+
+    def sorted(self):
+        """Utility function for sort_file_tabs_alphabetically()."""
+        for i in range(0, self.tabs.tabBar().count()-1):
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py]: Please add spaces around the `-` here.

File: spyder/plugins/editor/widgets/editor.py [hunk 6]
```diff
@@ -1568,6 +1571,22 @@ def close_all_but_this(self):
         for i in range(0, self.get_stack_count()-1  ):
             self.close_file(0)
 
+    def sort_file_tabs_alphabetically(self):
+        """Sort open tabs alphabetically."""
+        while self.sorted() is False:
+            for i in range(0, self.tabs.tabBar().count()):
+                if(self.tabs.tabBar().tabText(i) >
+                   self.tabs.tabBar().tabText(i+1)):
+                    self.tabs.tabBar().moveTab(i, i+1)
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py]: Also add spaces around the `+` here.

File: spyder/plugins/editor/widgets/editor.py (hunk 3 above)
Comment (CAM-Gerlach) [reply spyder/plugins/editor/widgets/editor.py]: More specifically, what @ccordoba12 described is our project's style convention for this situation.

**Comment by jnsebgosselin** [comment]
@ccordoba12 I think this fixes Issue https://github.com/spyder-ide/spyder/issues/5911

**Comment by ccordoba12** [comment]
Thanks for noticing it :+1:

**Pull request merged**